hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoCreateHandler.java

```diff
@@ -111,15 +112,22 @@ public CryptoCreateHandler(
     @Override
     public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckException {
         final var op = txn.cryptoCreateAccountOrThrow();
+        validateTruePreCheck(
+                op.autoRenewPeriod() != null && op.autoRenewPeriod().seconds() > -1, INVALID_RENEWAL_PERIOD);
+        if (op.hasShardID()) {
```

Comment:
Shard and realm shouldn't be in pure checks.
Those aren't "the node failed" issues; they're user error which should be checked in handle.
autorenew period also should be in handle.

The general path (once we avoid catastrophic failures) is to get transactions to handle so we succeed in charging the payer if anything else is wrong. The payer on the transaction is responsible for submitting a transaction that can be processed, and we want to hold that payer responsible as much as possible.

Reply:
Agreed with @netopyr

Reply:
If we need a temporary check, we could just check that realm and shard are exactly equal to `0`, given we only support the `0` shard and `0` realm currently.

`autoRenwePeriod` _should_ allow null (if we have trouble with null in that field it's possibly a bug), but if set I can see checking it's `>= 0`.

We definitely don't want to be checking for a non-null `newRealmAdminKey` as that's only for _creating a new realm_ and _should not_ be set.

Reply:
Thanks @jsync-swirlds. Have addressed now.